src/connection-manager/dialer/index.ts

```diff
@@ -171,10 +175,21 @@ export class DefaultDialer implements Startable, Dialer {
 
     log('creating dial target for %p', peerId)
 
+    if (Array.isArray(multiaddr)) {
+      // Return the first successful connection
+      return (await Promise.all(multiaddr.map(async (addr) => await this.createDialTarget(peerId, addr, options))))[0]
```

Comment:
We need to abort the creation of any other connections and/or close them, otherwise this will leak memory and sockets.